package/tests/test_api/test_rest/test_responses.py

```diff
@@ -794,15 +806,220 @@ def test_get_non_existing_pipeline(self, client):
+        filepath = str(Path(__file__).parent.parent)
         api_app = apps.create_api_app_from_file(filepath)
         client = TestClient(api_app)
         response = client.get("/")
         assert response.status_code == 200
+
+
+class TestPackageCompatibilities:
+    def test_get_package_compatibilities_response_compatible(self, mocker):
+        expected_version = "2023.9.1"
+        mocker.patch(
+            "kedro_viz.api.rest.responses.get_package_version",
+            return_value=expected_version,
+        )
+        response = get_package_compatibilities_response()
+        assert response.package_name == "fsspec"
+        assert response.package_version == expected_version
+        assert response.is_compatible is True
+
+    def test_get_package_compatibilities_response_incompatible(self, mocker):
+        expected_version = "2023.8.1"
+        mocker.patch(
+            "kedro_viz.api.rest.responses.get_package_version",
+            return_value=expected_version,
+        )
+        response = get_package_compatibilities_response()
+        assert response.package_name == "fsspec"
+        assert response.package_version == expected_version
+        assert response.is_compatible is False
+
+
+class TestEnhancedORJSONResponse:
+    @pytest.mark.parametrize(
+        "content, expected",
+        [
+            (
+                {"key1": "value1", "key2": "value2"},
+                b'{\n  "key1": "value1",\n  "key2": "value2"\n}',
+            ),
+            (["item1", "item2"], b'[\n  "item1",\n  "item2"\n]'),
+        ],
+    )
+    def test_encode_to_human_readable(self, content, expected):
+        result = EnhancedORJSONResponse.encode_to_human_readable(content)
+        assert result == expected
+
+    @pytest.mark.parametrize(
+        "file_path, response, encoded_response",
+        [
+            (
+                "test_output.json",
+                {"key1": "value1", "key2": "value2"},
+                b'{"key1": "value1", "key2": "value2"}',
+            ),
+        ],
+    )
+    def test_write_api_response_to_fs(
+        self, file_path, response, encoded_response, mocker
+    ):
+        mock_encode_to_human_readable = mocker.patch(
+            "kedro_viz.api.rest.responses.EnhancedORJSONResponse.encode_to_human_readable",
+            return_value=encoded_response,
+        )
+        with patch("kedro_viz.api.rest.responses.fsspec.filesystem") as mock_filesystem:
+            mock_remote_fs = mock_filesystem.return_value
+            mock_remote_fs.open.return_value.__enter__.return_value = Mock()
+            write_api_response_to_fs(file_path, response, mock_remote_fs)
+            mock_remote_fs.open.assert_called_once_with(file_path, "wb")
+            mock_encode_to_human_readable.assert_called_once()
+
+    @pytest.mark.parametrize(
+        "main_loc, expected_default_response",
+        [
+            ("/main", {"test": "json"}),
+        ],
+    )
```

Comment:
You've parameterised this test and several of the below, but as far as I can tell you're only passing one set of values right? In that case it might actually be more readable if you don't parameterise the tests where you don't have multiple test values.